src/OpenLoco/src/Paint/PaintBuilding.cpp

```diff
@@ -73,7 +73,7 @@ namespace OpenLoco::Paint
             }
             ImageId scaffImage = baseScaffImage.withIndexOffset(scaffImages.back);
             auto segmentImageOffset = imageOffset;
-            for (int8_t remainingHeight = totalSectionHeight; remainingHeight > 0; remainingHeight -= segmentHeight, segmentImageOffset.z += segmentHeight)
+            for (int16_t remainingHeight = totalSectionHeight; remainingHeight > 0; remainingHeight -= segmentHeight, segmentImageOffset.z += segmentHeight)
```

Comment:
Wouldn't it be better to use coord_t? If we ever raise limits we don't want to run around all code really.